server/serverAPI/controller/AllergenController.ts

```diff
@@ -0,0 +1,158 @@
+import { Request, Response } from "express";
+import IDatabase from "../../database/IDatabase";
+import { ResponseCodes } from "../../utils/ResponseCodes";
+import IBaseIngredient from "../model/ingredient/IBaseIngredient";
+
+import IInventoryIngredient from "../model/ingredient/IInventoryIngredient";
+import BaseIngredientSchema from "../model/ingredient/requestSchema/BaseIngredientSchema";
+import IUser from "../model/user/IUser";
+import BaseIngredientController from "./BaseIngredientController";
```

Comment:
You can fix the old ones when you have time, but starting now, can you group code that is external to the project and code that you wrote (with related stuff together or alphabetically)?
```js
// external
import { Request, Response } from "express";

// utils
import { ResponseCodes } from "../../utils/ResponseCodes";

// interfaces
import IBaseIngredient from "../model/ingredient/IBaseIngredient";
import IDatabase from "../../database/IDatabase";
import IInventoryIngredient from "../model/ingredient/IInventoryIngredient";
import IUser from "../model/user/IUser";

// schemas
import BaseIngredientSchema from "../model/ingredient/requestSchema/BaseIngredientSchema";

// controllers
import BaseIngredientController from "./BaseIngredientController";
```
Note, you don't need the comments I added, but if you want, you can add them